identity-account/src/identity/identity_create.rs

```diff
@@ -47,6 +49,16 @@ impl IdentityCreate {
     self.network = Some(value.into());
     self
   }
+
+  /// Sets the [SecretKey] for the Identity creation.
+  /// This needs to be the correct number of bytes for the respective [KeyType].
+  ///
+  /// For [KeyType::Ed25519] the key needs to be 32 bytes in size.
+  #[must_use]
+  pub fn secret_key(mut self, value: SecretKey) -> Self {
+    self.secret_key = Some(value);
+    self
+  }
```

Comment:
`SecretKey` seems specific to `KeyType::Ed25519`, which will be a problem since other methods may use different representations, e.g. Merkle keys use `KeyCollection` / `Box<[SecretKey]>`. We could use something like an enum with optional associated data as the `key_type` instead.

Since the `Account` lacks support for Merkle key collections in general this is fine for now, just noting that it will need a breaking change in the future.